tests/queries/tests.py

```diff
@@ -2235,6 +2235,29 @@ def test_ticket_18414_distinct_on(self):
         self.assertTrue(Article.objects.distinct("name")[1:2].exists())
         self.assertFalse(Article.objects.distinct("name")[2:3].exists())
 
+    def test_ticket_24296(self):
+        """
+        Don't clear select fields only for distinct() + limit + exists() combination
+        """
+
+        with CaptureQueriesContext(connection) as captured_queries:
+            Article.objects.exists()
+            Article.objects.distinct().exists()
+            Article.objects.distinct()[1:3].exists()
+        (
+            exist_qr_str,
+            distinct_exist_qr_str,
+            distinct_sliced_exist_qr_str,
+        ) = captured_queries
+
+        # Model's fields are present only in the sliced query
+        self.assertNotIn("name", exist_qr_str["sql"])
+        self.assertNotIn("created", exist_qr_str["sql"])
+        self.assertNotIn("name", distinct_exist_qr_str["sql"])
+        self.assertNotIn("created", distinct_exist_qr_str["sql"])
+        self.assertIn("name", distinct_sliced_exist_qr_str["sql"])
+        self.assertIn("created", distinct_sliced_exist_qr_str["sql"])
```

Comment:
This won't work on all backends, we need to use `connection.ops.quote_name()` for identifiers.